### ISSUE TITLE
Warning on Elixir v1.14

### ISSUE THREAD

**Issue by thbar** [maennchen/crontab#102]
I'm doing a bit of preliminary testing with https://github.com/elixir-lang/elixir/releases/tag/v1.14.0-rc.1, and noticed the following warning, so I'm opening an issue to report it :

```
==> crontab
Compiling 7 files (.ex)
warning: Application.get_env/3 is discouraged in the module body, use Application.compile_env/3 instead
  lib/crontab/scheduler.ex:21: Crontab.Scheduler

Generated crontab app
```

Apparently not fixed on `master`:

https://github.com/jshmrtn/crontab/blob/c7641a424abc1191c3bf91ace369350097eb4890/lib/crontab/scheduler.ex#L21

**Comment by maennchen** [comment]
Fixed by 81bd15abaa954afc8e8f9d1fd5b295fb3bf59139

**Comment by sergchernata** [comment]
Looks like the warning is still being triggered:

```
Compiling 7 files (.ex)
    warning: Application.get_env/3 is discouraged in the module body, use Application.compile_env/3 instead
    │
 21 │     @max_runs Application.get_env(:crontab, :max_runs, 10_000)
    │                           ~
    │
    └─ lib/crontab/scheduler.ex:21:27: Crontab.Scheduler (module)

```

**Comment by thbar** [comment]
Same here! Maybe we can reopen a new issue @sergchernata?

**Comment by Hermanverschooten** [comment]
It seems this has been resolved in #116, but there has not been a new release?

**Comment by a-maze-d** [comment]
I was looking into this issue, because I got the warning too. It looks like the `1.1.14` release should have fixed the issue (mainline has the new code), and was [committed](https://github.com/maennchen/crontab/blob/main/lib/crontab/scheduler.ex#L16) in 2023:
```elixir
  @max_runs Application.compile_env(:crontab, :max_runs, 10_000)
```
But the latest release (23rd Oct 2024) does contain the [old code](https://github.com/maennchen/crontab/blob/v1.1.14/lib/crontab/scheduler.ex#L16):
```elixir
 # TODO: Remove if when requiring Elixir 1.10 + only
  if function_exported?(Application, :compile_env, 3) do
    @max_runs Application.compile_env(:crontab, :max_runs, 10_000)
  else
    # credo:disable-for-next-line Credo.Check.Warning.ApplicationConfigInModuleAttribute
    @max_runs Application.get_env(:crontab, :max_runs, 10_000)
  end
```

It's a bit unclear to me why the latest release did not pick up the changes on `main`. It looks like the release was not made from `main` :-(

Thus, only a new release should be required to fix the warning (as long as it's made from `main` :-)

**Comment by maennchen** [comment]
@a-maze-d `v1.1.14` was a bug release that only contained #132 backported since the current `main` is not in shape to be released.

Once #133 is done, I'll do a new release including everything.

**Comment by a-maze-d** [comment]
Cool. Thanks
